### PR TITLE
populate repository endpoint for legacy powershell handler.

### PR DESCRIPTION
**Pull request by TingluoHuang**

### CODE REVIEW

File: src/Agent.Worker/Handlers/LegacyPowerShellHandler.cs [hunk 1]
```diff
@@ -389,6 +390,27 @@ protected virtual void AddLegacyHostEnvironmentVariables(string scriptFile, stri
                 }
             }
 
+            var defaultRepoName = ExecutionContext.Variables.Get(Constants.Variables.Build.RepoName);
+            var defaultRepoType = ExecutionContext.Variables.Get(Constants.Variables.Build.RepoProvider);
+            if (!string.IsNullOrEmpty(defaultRepoName))
+            {
+                var defaultRepo = ExecutionContext.Repositories.FirstOrDefault(x => String.Equals(x.Properties.Get<string>(RepositoryPropertyNames.Name), defaultRepoName, StringComparison.OrdinalIgnoreCase));
```
Comment (ericsciple) [inline src/Agent.Worker/Handlers/LegacyPowerShellHandler.cs]: think about this for multi-checkout

Comment (ericsciple) [reply src/Agent.Worker/Handlers/LegacyPowerShellHandler.cs]: why not use the alias?

File: src/Microsoft.VisualStudio.Services.Agent/Util/VarUtil.cs [hunk 2]
```diff
@@ -63,6 +87,28 @@ public static void ExpandEnvironmentVariables(IHostContext context, IDictionary<
             ExpandValues(context, source, target);
         }
 
+        public static JToken ExpandValues(IHostContext context, IDictionary<string, string> source, JToken target)
+        {
+            var mapFuncs = new Dictionary<JTokenType, Func<JToken, JToken>>
+            {
+                {
+                    JTokenType.String,
```
Comment (ericsciple) [inline src/Microsoft.VisualStudio.Services.Agent/Util/VarUtil.cs]: null?

File: src/Agent.Worker/JobRunner.cs [hunk 3]
```diff
@@ -198,18 +198,22 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
...
-
-                    foreach (var expandedProperty in expandProperties)
+                    // expand workspace mapping
+                    var mappings = repository.Properties.Get<JToken>(Pipelines.RepositoryPropertyNames.Mappings);
+                    if (mappings != null)
                     {
-                        repository.Properties.Set<JToken>(expandedProperty.Key, JsonUtility.FromString<JToken>(expandedProperty.Value));
+                        mappings = jobContext.Variables.ExpandValues(target: mappings);
+                        mappings = VarUtil.ExpandEnvironmentVariables(HostContext, target: mappings);
+                        repository.Properties.Set<JToken>(Pipelines.RepositoryPropertyNames.Mappings, mappings);
```
Comment (ericsciple) [inline src/Agent.Worker/JobRunner.cs]: do we need to trace anything?

Comment (ericsciple) [reply src/Agent.Worker/JobRunner.cs]: should varutil trace if it replaces something? (tracing for free)

Comment (ericsciple) [reply src/Agent.Worker/JobRunner.cs]: ...might be too confusing

**Comment by ericsciple** [comment]
looks good

File: src/Microsoft.VisualStudio.Services.Agent/Util/VarUtil.cs (hunk 2 above)
Comment (TingluoHuang) [reply src/Microsoft.VisualStudio.Services.Agent/Util/VarUtil.cs]: JTokenType.Null will just return the original value

File: src/Agent.Worker/JobRunner.cs (hunk 3 above)
Comment (TingluoHuang) [reply src/Agent.Worker/JobRunner.cs]: VarUtil has built-in verbose tracing.

**Pull request merged**